packages/dds/tree/src/test/feature-libraries/chunked-forest/chunkEncodingEndToEnd.spec.ts

```diff
@@ -374,5 +384,52 @@ describe("End to end chunked encoding", () => {
 			const jsonableTree = mapCursorField(chunk.cursor(), jsonableTreeFromCursor);
 			assert.deepEqual([initialTree], jsonableTree);
 		});
+
+		it("Initializing tree creates uniform chunks with encoded identifiers", async () => {
+			const factory = new SharedTreeFactory({
+				jsonValidator: typeboxValidator,
+				forest: ForestType.Optimized,
+			});
+			const provider = new TestTreeProviderLite(1, factory, true);
+
+			// Workaround for getting the same id values from the TestTreeProviderLite's deterministic id compressor
+			const random = makeRandom(0xdeadbeef);
+			const deterministicIdCompressor = createIdCompressor(random.uuid4() as SessionId);
+			const stableId = deterministicIdCompressor.decompress(
+				deterministicIdCompressor.generateCompressedId(),
+			);
+
+			class TreeWithIdentifier extends schemaFactory.object("treeWithIdentifier", {
+				identifier: schemaFactory.identifier,
+			}) {}
+			const view = provider.trees[0].viewWith(
+				new TreeViewConfiguration({
+					schema: TreeWithIdentifier,
+				}),
+			);
+			view.initialize({});
+			const forest = view.checkout.forest;
+			assert(forest instanceof ChunkedForest);
+			const uniformChunk = forest.roots.fields.get(rootFieldKey)?.at(0);
+			assert(uniformChunk instanceof UniformChunk);
+			const chunkValues = uniformChunk.values;
+			assert.deepEqual(chunkValues, [deterministicIdCompressor.recompress(stableId)]);
+
+			// When getting the value from the cursor, check that the value is unencoded string
+			const jsonableTree = mapCursorField(uniformChunk.cursor(), jsonableTreeFromCursor);
+			assert.deepEqual(jsonableTree, [
+				{
+					fields: {
+						identifier: [
+							{
+								type: "com.fluidframework.leaf.string",
+								value: stableId,
+							},
+						],
+					},
+					type: "com.example.treeWithIdentifier",
+				},
+			]);
+		});
```

Comment:
This is a great test.

I hesitate to suggest it since doing so would test other layers of abstraction that don't really belong here, but it would be super east to test that the public API works properly, and that is the suer facing thing so its pretty high value so I think its worth it.

Asserting that view.root.identifier === stableId and Tree.shortId === the compressed id would cover the full end to end, and its just two more one line asserts.

Reply:
Added two more asserts to cover the full end to end scenario